core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -232,7 +232,7 @@ object StorageTool extends Logging {
       }
     })
     if (unformattedDirectories.isEmpty) {
-      throw new TerseFailure("All of the log directories are already formatted.")
+      stream.println("All of the log directories are already formatted.")
```

Comment:
Fair enough.  We can make this non-fatal (it only happens when --ignore-formatted is passed, anyway...)